satpy/scene.py

```diff
@@ -150,6 +150,12 @@ def __init__(self, filenames=None, reader=None, filter_parameters=None, reader_k
         self.readers = self.create_reader_instances(filenames=filenames,
                                                     reader=reader,
                                                     reader_kwargs=reader_kwargs)
+
+        if filenames and not self.available_dataset_ids(reader_name=reader):
```

Comment:
The only problem I could not resolve is: If no file matches the ``filter_parameters`` the exception is triggered mistakenly. But I would expect ``create_reader_instances`` to throw an exception in this case? Previously you would run into the ``Start time unknown until files are selected`` error.

Reply:
Sorry I'm a little confused. So if none of the files match (due to filter_parameters or other reasons) then `create_reader_instances` isn't raising an exception or it is?

Also, you should probably remove the `reader_name=reader` part since filenames can be a dictionary mapping `reader: [filenames]`. Just to check if there are any available datasets for any possible readers.

Reply:
``create_reader_instances`` does _not_ raise an exception in this case. Instead the subsequent call to ``_compute_metadata_from_readers``  causes the following exception:

```
Traceback (most recent call last):
  File "test.py", line 21, in <module>
    end_time=datetime.datetime(2010, 1, 2)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 159, in __init__
    self.attrs.update(self._compute_metadata_from_readers())
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 177, in _compute_metadata_from_readers
    for x in self.readers.values())
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 177, in <genexpr>
    for x in self.readers.values())
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/yaml_reader.py", line 296, in start_time
    raise RuntimeError("Start time unknown until files are selected")
RuntimeError: Start time unknown until files are selected
```

Reply:
So maybe `load_readers` (in `satpy/readers/__init__.py` which is called by `create_reader_instances`) should change:

```python
    if not reader_instances:
        raise ValueError("No supported files found")
```

to something like:

```python
if not reader_instances or all(reader.available_dataset_ids for reader in reader_instances):
    raise ValueError("No supported files found")
```

Then I think you could remove the `scene.py` change. If the error messages would be clearer I suppose the above if statement could be split in to `if not reader_instances:` and `elif all(...):` so that it is more clear "you gave valid files, but not enough, or not useable ones" or something like what you had.

Reply:
Good idea! I moved the check & exception to ``load_readers``

Reply:
You said you moved this, but it still exists and doesn't seem to exist in `load_readers`. Forgot to push something?

Reply:
Strange, my local and remote branch are up-to-date. Also, the ``Files Changed`` tab shows the diffs I commited, especially no changes to ``scene.py``. Maybe if I fix the typo and push again, it will be updated...